File: Samples for Solid Edge/[VB]-Samples for Solid Edge/VB/SolidEdge.Assembly.BOM/BomItem.vb
Imports System.Text

Namespace SolidEdge.Assembly.BOM
	Public Class BomItem
		Inherits MarshalByRefObject

		Public Sub New(ByVal fullName As String)
			Me.FullName = fullName
			FileName = System.IO.Path.GetFileName(fullName)
			Quantity = 1
		End Sub

		Public Level As Integer
		Public DocumentNumber As String
		Public Revision As String
		Public Title As String
		Public Quantity As Integer
		Public FullName As String
		Public FileName As String
		Public IsSubassembly As Boolean
	End Class
End Namespace

File: Samples for Solid Edge/[C#]-Samples for Solid Edge/C#/SolidEdge.Assembly.BOM/BomItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SolidEdge.Assembly.BOM
{
    public class BomItem : MarshalByRefObject
    {
        public BomItem(string fullName)
        {
            FullName = fullName;
            FileName = System.IO.Path.GetFileName(fullName);
            Quantity = 1;
        }

        public int Level;
        public string DocumentNumber;
        public string Revision;
        public string Title;
        public int Quantity;
        public string FullName;
        public string FileName;
        public bool IsSubassembly;
    }
}
